src/ocean/THCM.C

```diff
@@ -3018,3 +2353,764 @@ Teuchos::RCP<const Epetra_MultiVector> THCM::getNullSpace()
     return nullSpace;
 }
 
+//=============================================================================
+Teuchos::ParameterList&
+THCM::setDefaultInitParameters(Teuchos::ParameterList& params)
+{
+    params.get("Rho Mixing",true);
+    params.get("Problem Description","Unnamed");
+    params.get("Grid Stretching qz", 1.0);
+    params.get("Topography", 1);
+    params.get("Flat Bottom", false);
+    params.get("Read Land Mask", false); //== false in experiment0
+    params.get("Land Mask","no_mask_specified");
+
+    params.get("Read Salinity Perturbation Mask",false);
+    params.get("Salinity Perturbation Mask", "no_mask_specified");
+
+    params.get("Coriolis Force", 1);
+    params.get("Forcing Type", 0);
+
+    params.get("Wind Forcing Data", "wind/trtau.dat");
+    params.get("Temperature Forcing Data", "levitus/new/t00an1");
+    params.get("Salinity Forcing Data", "levitus/new/s00an1");
+
+    params.get("Time Dependent Forcing", false);
+
+    return THCM::setDefaultParameters(params);
+}
+
+Teuchos::ParameterList&
+THCM::setDefaultParameters(Teuchos::ParameterList& params)
+{
+    params.get("Global Grid-Size n", 16);
+    params.get("Global Grid-Size m", 16);
+    params.get("Global Grid-Size l", 16);
+
+    // default: north atlantic
+    params.get("Global Bound xmin", 286.0);
+    params.get("Global Bound xmax", 350.0);
+    params.get("Global Bound ymin", 10.0);
+    params.get("Global Bound ymax", 74.0);
+
+    params.get("Periodic"  , false);
+    params.get("Depth hdim", 4000.0);
+    params.get("Inhomogeneous Mixing",0);
+    params.get("Mixing",1);
+    params.get("Taper",1);
+    params.get("Linear EOS: alpha T", 1.0e-4);
+    params.get("Linear EOS: alpha S", 7.6e-4);
+    params.get("Restoring Temperature Profile",1);
+    params.get("Restoring Salinity Profile",1);
+    params.get("Local SRES Only", false);
+    params.get("Salinity Integral Sign", -1);
+    params.get("Levitus T", 1);
+    params.get("Levitus S", 1);
+    params.get("Levitus Internal T/S",false);
+    params.get("Compute salinity integral", true);
+    params.get("Coupled Temperature", 0);
+    params.get("Coupled Salinity", 0);
+    params.get("Coupled Sea Ice Mask", 1);
+    params.get("Fix Pressure Points", false);
+    params.get("Wind Forcing Type",2);
+    params.get("Scaling","THCM");
+
+    Teuchos::ParameterList& startParams = params.sublist("Starting Parameters");
+    /*
+    for (int i=0; i<= _NPAR_ + _NPAR_TRILI; i++)
```

Comment:
The "default" values returned here by `defaultParameter` aren't, well, the actual defaults, because uncommenting this code breaks most of the tests. We should probably fix these default values, but I'm unsure where the current values are set.

Reply:
They are in `usr.F90`